### ISSUE TITLE
Cart and Change.js bug - Ajax cart doesn't update quantity after Shopify script uses .split, and removes item if already split through normal cart

### ISSUE THREAD

**Issue by jonathan-bird** [Shopify/Timber#579]
# Issue 
There's a bug we found that the AJAX Cart doesn't update the quantity of an item if it's going through a split in Shopify script. All other products in cart can update quantity, except for this one. When I increase quantity to 3 of this item, it splits so I have 1 item with 100% discount, and 2 with normal pricing (both different keys, same variant id), and then I open AJAX cart and update the quantity of the normal priced item, it removes the item from the cart all together. I then try to update quantity of free item, and it just sends request through change.js of line: 1, quantity: 2, and returns response with quantity 1 and no more items. 

So basically, it works through the Liquid cart, but not the AJAX Change.js and Cart.js

# Replicate

1. Through Shopify script, split an item after 1 and add 3 to cart (through liquid cart, non ajax cart), see that it splits correctly.

Split example: 
`.split example

if original_line_item.quantity >= 1
  new_line_item = original_line_item.split(take: 1)
  new_line_item.change_line_price(Money.new(cents: 500), message: "Third hat for 5 dollars")
  cart.line_items << new_line_item
end`

2. Open ajax cart, see 2 different items, 
3. Click update quantity of either items. Notice that the item gets removed from the cart
